crud/common/utils.lua

```diff
@@ -148,6 +151,23 @@ function utils.extract_key(tuple, key_parts)
     return key
 end
 
+function utils.extract_jsonpath_keys(tuple, index_parts, key_def)
+    key_def = key_def or keydef_lib.new(index_parts)
```

Comment:
I don't like such logic. Why can't you use just key_def instance?

Anyway such logic won't prevent a case when you'll recreate key_def instance if the third argument is skipped.